src/components/pages/account/InvestmentChart.tsx

```diff
@@ -50,61 +52,161 @@ export default function InvestmentChart({
   const numStocksData: { x: number, y: number }[] = [];
   const valueData: { x: number, y: number }[] = [];
   const dates = intervalToDates(interval);
+  const isCurrencyInvestment = investment.account.commodity.namespace === 'CURRENCY';
 
-  dates.forEach(x => {
-    investment.processSplits(x);
-    numStocksData.push({
-      x: x.startOf('month').toMillis(),
-      y: investment.quantity.toNumber(),
+  investment.splits
+    .filter(s => interval.contains(s.transaction.date) && !InvestmentAccount.isDividend(s))
+    .forEach(s => {
+      investment.processSplits(s.transaction.date);
+      numStocksData.push({
+        x: s.transaction.date.toMillis(),
+        y: !isCurrencyInvestment ? investment.quantity.toNumber() : investment.cost.toNumber(),
+      });
+
+      const price = pricesMap.getInvestmentPrice(
+        investment.account.commodity.mnemonic,
+        s.transaction.date,
+      );
+
+      if (price) {
+        valueData.push({
+          x: s.transaction.date.toMillis(),
+          y: investment.quantity.toNumber() * price.value,
+        });
+      }
```

Comment:
so basically we are changing the x axis data points. Now we show a point every time there is a buy/sell/split movement plus the last day of each month (if it doesn't exist already).